### ISSUE TITLE
caffeine fallback icons clash with other packages

### ISSUE THREAD

**Issue by fossfreedom** [solus-project/budgie-desktop#1529]
**Operating System**: Debian Buster
**Budgie version**: git master cb35f5b4f1c71d8dfa3aa8f78298b3825ce85cc2

**The issue encountered**

The fallback icons caffeine-cup-full/empty.svg are installed in the hicolor folder.  Unfortunately, other desktop environments have similar caffeine based capabilities which similarly also install the same named icons into the hicolor folder.

When users who run more than one DE attempt to install budgie - budgie fails to install due to this clash

I guess the best way here is to maybe install the caffeine icons specifically - or more generally the budgie icons that were destined for .../icons/hicolor to a subfolder under budgie-desktop and use the append_search_path https://valadoc.org/gtk+-3.0/Gtk.IconTheme.append_search_path.html to ensure budgie based fallback icons are found in this budgie specific location.

Thoughts?  Best place to do the "IconTheme.get_default() / IconTheme.append_search_path("....../budgie-desktop/iconspath") - I guess some constructor which is only initialised once.

**Comment by yursan9** [comment]
I don't think we should muck around hicolor's directory hierarchy. We can just change the name, and if current Icon Theme doesn't support the Icon we can use the fallback.

Can I know the name of Desktop Environment that using the same icon's name (for research)?
I admit the fallback icon I made is unsightly, so I don't mind if another application overwrite the icons. :grin:

**Comment by fossfreedom** [comment]
@yursan9 yeah - reading the GTK docs further it says that the append_search_path stuff is for legacy only.

Any GTK environment (I was using the default XFCE in debian buster + caffeine package when I discovered this issue) this applies to when the caffeine package.  Also Unity and Budgie under Ubuntu that has caffeine installed.

Under debian packaging, one package cannot overwrite another package contents even down to individual file level - hence the conflict here.  

As a packager I have a number of choices here - none are great

1. Hold a local only patch that installs the upstream caffeine icons with "budgie" names and change the caffeine code to use these new names.  Downside - users cannot easily change the icon - and you are left using the permanent fallback icons.  Personally don't like that idea.

2. When installing don't install the fallback icons.  Users are forced to use an icon-theme that has caffeine icons ... or to fix it themselves when an icon-theme doesnt have those icons.  Not great... and I guess Debian and Ubuntu users will start to raise issues here.

3. Similar to 2 - but bundle the caffeine package when installing budgie-desktop.  This just really feels unnecessary bloat.

4. Like 1 but additionally work out if an icon theme doesnt have those icons and switch to using the budgie equivalents.  Messy and not clean.

I've played around with idea 4 - https://github.com/UbuntuBudgie/budgie-desktop/commits/caffeine - the two commits there.

Currently toying around with idea 2 or 4 but would be interested in other thoughts.
